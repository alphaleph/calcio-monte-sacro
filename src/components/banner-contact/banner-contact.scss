@import '../../pages/globals.scss';

.banner-contact-container {
    position: relative;
    border-top: 1px solid $holy-yellow-team;
    border-bottom: 1px solid $holy-yellow-team;
    background: $black-main;
}

.banner-contact-textbox {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 3rem 0;
}

.banner-contact-header {
    color: $white-main;
    font-weight: 600;
    text-align: center;
    font-size: 1.6rem;
    margin: 2rem 0;
}

.banner-contact-text {
    color: $white-main;
    // font-weight: bold;
    text-align: center;
    margin-bottom: 2rem;
    font-size: 1.3rem;
}

.banner-contact-link {
    color: $holy-yellow-team;
}

.banner-contact-link:focus,
.banner-contact-link:hover {
    text-decoration: underline;
    color: $holy-yellow-team;
}

@media screen and (min-width: 550px) {
    .banner-contact-header {
        font-size: 2rem;
    }
    .banner-contact-text {
        font-size: 1.7rem;
    }
}