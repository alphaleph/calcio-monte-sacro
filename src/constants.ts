// Site Constants
export const SITE_URL = 'https://www.calciomontesacro.com';
export const EMAIL_ADDR = `&#099;&#097;&#108;&#099;&#105;&#111;&#109;&#111;&#110;&#116;&#101;&#115;&#097;&#099;&#114;&#111;&#064;&#103;&#109;&#097;&#105;&#108;&#046;&#099;&#111;&#109`;
export const NEWS_URL = SITE_URL + '/news';

// CMS Constants
export const HOME_VENUE = `Parco Sannazzaro`;
export const TEAM_ABRV = `CMS`;
export const MATCH_EN = `Match`;
export const MATCH_IT = `Partita`;
export const PRACTICE_EN = `Practice`;
export const PRACTICE_IT = `Allenamento`;

// openModal Constants
export const NONE = 'NONE';
export const TERMS_AND_CONDITIONS = 'TERMS_AND_CONDITIONS';
export const PRIVACY = 'PRIVACY';
export const COOKIES = 'COOKIES';
