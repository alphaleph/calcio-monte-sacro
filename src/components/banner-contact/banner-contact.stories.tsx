import React from 'react';
import { storiesOf } from '@storybook/react';
import { BannerContact } from './banner-contact';
import '../../pages/index.scss';

storiesOf('BannerContact', module).add('default', () => <BannerContact />);
