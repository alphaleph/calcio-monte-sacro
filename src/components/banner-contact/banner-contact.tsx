import React from 'react';
import './banner-contact.scss';
import { EMAIL_ADDR, LOCATION_URL } from '../../constants';

export const BannerContact = () => {
    return (
        <section className="banner-contact-container hero is-large has-background-granite-grey-team">
            <div className="banner-contact-textbox">
                <h2 className="banner-contact-header">CONTACT US</h2>
                <p className="banner-contact-text">
                    Meet us at
                    <br />
                    <a className="banner-contact-link" href={LOCATION_URL}>
                        Parco Sannazzaro, 00141 Roma RM, Italy
                    </a>
                </p>
                <p className="banner-contact-text">
                    Email us at
                    <br />
                    <a
                        className="banner-contact-link"
                        href={`mailto:${EMAIL_ADDR}`}
                    >
                        {EMAIL_ADDR}
                    </a>
                </p>
            </div>
        </section>
    );
};
