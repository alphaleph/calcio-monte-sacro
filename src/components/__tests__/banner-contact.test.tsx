import React from 'react';
import { BannerContact } from '../banner-contact/banner-contact';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { EMAIL_ADDR, HOME_LOCATION, LOCATION_URL } from '../../constants';

const title = 'CONTACT US';
const emailLink = `mailto:${EMAIL_ADDR}`;

describe('Banner Contact', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<BannerContact />);
        expect(getByText(title)).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<BannerContact />);
        expect(getByText(title)).toBeVisible();
        expect(getByText(HOME_LOCATION)).toBeVisible();
        expect(getByText(HOME_LOCATION)).toHaveAttribute('href', LOCATION_URL);
        expect(getByText(EMAIL_ADDR)).toBeVisible();
        expect(getByText(EMAIL_ADDR)).toHaveAttribute('href', emailLink);
    });

    it('has not changed', () => {
        const { asFragment } = render(<BannerContact />);
        expect(asFragment()).toMatchSnapshot();
    });
});
