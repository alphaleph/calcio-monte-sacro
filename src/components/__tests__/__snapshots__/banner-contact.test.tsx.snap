// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Contact has not changed 1`] = `
<DocumentFragment>
  <section
    class="banner-contact-container hero is-large has-background-granite-grey-team"
  >
    <div
      class="banner-contact-textbox"
    >
      <h2
        class="banner-contact-header"
      >
        CONTACT US
      </h2>
      <p
        class="banner-contact-text"
      >
        Meet us at
        <br />
        <a
          class="banner-contact-link"
          href="https://www.google.com/maps/place/Parco+Sannazzaro/@41.9422582,12.5402792,17z/data=!3m1!4b1!4m5!3m4!1s0x132f64080252da0b:0x28b23f52b87f2d63!8m2!3d41.9422582!4d12.5424679"
        >
          Parco Sannazzaro, 00141 Roma RM, Italy
        </a>
      </p>
      <p
        class="banner-contact-text"
      >
        Email us at
        <br />
        <a
          class="banner-contact-link"
          href="mailto:calciomontesacro@gmail.com"
        >
          calciomontesacro@gmail.com
        </a>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
